js-pkg/apps/demos/src/pages/voxel.tsx

```diff
@@ -7,58 +7,49 @@ import { Vector3, Vector3Tuple } from 'three'
 import { CompactPicker } from 'react-color'
 
 const DIM = 15
-const TRANSITION_DURATION_MS = 0.3
+const TRANSITION_RATE = 0.07
 
 interface Voxel {
   position: Vector3Tuple
   color: any
   opacity: number
 }
 
-interface MidTransition {
-  startPosition: Vector3Tuple
-  delta: Vector3Tuple
-  start: number | null
-}
-
 function MovingVoxel(props: { voxel: Voxel; name?: string }) {
   const [position, setPosition] = useState<Vector3Tuple>([0, 0, 0]) // this gets set before the first paint
-  const transitionRef = useRef<MidTransition | null>(null)
+  const destPositionRef = useRef<Vector3Tuple | null>(null)
```

Comment:
I like this, I agree that it's cleaner code and a better look.

One thing I'm tempted to add is a minimum speed, so that instead of crawling towards the end it still “snaps” into place. Thoughts? Maybe it would be sufficient to just make it go a little faster.

Reply:
Maybe speeding it up a little would make it feel better?

Reply:
It still feels like the transition drags on too long, but I'm good with landing this and I can tweak it if it bothers me enough :)